From 6f7046f4894e0a95cbbc85ae62c6c0b4d95ac43c Mon Sep 17 00:00:00 2001
From: fundamental <mark.d.mccurry@gmail.com>
Date: Fri, 17 Oct 2014 11:30:50 -0400
Subject: [PATCH 0247/1936] Fix Possible TLSF Edge Case

This may be fixing a bug generated by undefined behavior elsewhere in the zyn
source, but it should have no negative consequences if it is left in
---
 tlsf/tlsf.c | 3 +++
 1 file changed, 3 insertions(+)

diff --git a/tlsf/tlsf.c b/tlsf/tlsf.c
index 3fb5ebd2..69963960 100644
--- a/tlsf/tlsf.c
+++ b/tlsf/tlsf.c
@@ -561,6 +561,9 @@ static block_header_t* block_locate_free(control_t* control, size_t size)
 	{
 		mapping_search(size, &fl, &sl);
 		block = search_suitable_block(control, &fl, &sl);
+		if(block && !block->size)
+			block = NULL;
+
 	}
 
 	if (block)
-- 
2.47.0

